File: app/src/androidTest/java/com/example/energy/AesGcmBenchmarkInstrumentedTest.java
package com.example.energy;

import android.app.Instrumentation;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.filters.LargeTest;
import androidx.test.platform.app.InstrumentationRegistry;

import com.example.energy.tests.BenchmarkTestUtils;

import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
@LargeTest
public class AesGcmBenchmarkInstrumentedTest {
    @Test
    public void runAESGCMRangeAndValidateCsv() {

        int messageSize = Integer.parseInt(
                InstrumentationRegistry.getArguments().getString("contentsize", "10")
        );
        var context = BenchmarkTestUtils.INSTANCE.ctx();
        var fileName = "test_aes_2p10_2p20.csv";

        var path = AesGcmBenchmark.INSTANCE.runRangeAndLog(
             context, messageSize, messageSize,15, fileName
        );
        assert(true);
        //BenchmarkTestUtils.INSTANCE.assertCsv(path);
    }
}

File: app/src/androidTest/java/com/example/energy/AesGcmBenchmarkInstrumentedTest.kt
package com.example.energy.tests

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.example.energy.AesGcmBenchmark
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class AesGcmBenchmarkInstrumentedTest {
    @Test
    fun runAESGCMRangeAndValidateCsv() {
        val path = AesGcmBenchmark.runRangeAndLog(
            context = BenchmarkTestUtils.ctx(),
            minPow = BenchmarkTestUtils.MIN_POW,
            maxPow = BenchmarkTestUtils.MAX_POW,
            fileName = "test_aesgcm_2p10_2p20.csv"
        )
        BenchmarkTestUtils.assertCsv(path)
    }
}
